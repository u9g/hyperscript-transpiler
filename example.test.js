// sum.test.js
// import { sum } from "./sum";
import { expect, test } from "vitest";
const { run } = require("./index");

test("basic 1", async () => {
  expect(
    await run(`
  on "worldLoad"
    exampleWorldLoad()
    exampleWorldLoad2()
  end

  on "quit"
    sendLeave()
  end
  `)
  ).toMatchSnapshot();
});

test("basic 2", async () => {
  expect(
    await run(`
    on "click" call myJavascriptFunction()
    `)
  ).toMatchSnapshot();
});

test("basic 3", async () => {
  expect(
    await run(`
    on "click"
      call prompt('age fn?')
      call it()
    end
    `)
  ).toMatchSnapshot();
});

test("basic 4", async () => {
  expect(
    await run(`
    log "Hello"
    log "World!"
    `)
  ).toMatchSnapshot();
});

test("basic 5", async () => {
  expect(
    await run(`
      log "hello" then log "world" then log "!"
    `)
  ).toMatchSnapshot();
});

test("basic 6", async () => {
  expect(
    await run(`
      on "click" log "hello" then log "world"
    `)
  ).toMatchSnapshot();
});

test("basic 7", async () => {
  expect(
    await run(`
      on "click" call helloWorld() then log result
    `)
  ).toMatchSnapshot();
});

test("basic 8", async () => {
  expect(
    await run(`
      set x to "5"
    `)
  ).toMatchSnapshot();
});

test("basic 9", async () => {
  expect(
    await run(`
      set x to apple
    `)
  ).toMatchSnapshot();
});

test("basic 9", async () => {
  expect(
    await run(`
      set x to 45
    `)
  ).toMatchSnapshot();
});

test("basic 10", async () => {
  expect(
    await run(`
      set x to {name : "Joe", age: 35}
    `)
  ).toMatchSnapshot();
});

test("basic 11", async () => {
  expect(
    await run(`
      on "keyup" log it's x
    `)
  ).toMatchSnapshot();
});

test("basic 11", async () => {
  expect(
    await run(`
      set y to \\ x -> x then call y(5)
    `)
  ).toMatchSnapshot();
});

test("basic 12", async () => {
  expect(
    await run(`
      set y to \\-> true
    `)
  ).toMatchSnapshot();
});

test("basic 13", async () => {
  expect(
    await run(`
      set y to {a: \\ x -> x} then call y's a('a')
    `)
  ).toMatchSnapshot();
});

test("basic 14", async () => {
  expect(
    await run(`
      set x to 10
      set y to 20
      set sumOfSquares to (x * x) + (y * y)
    `)
  ).toMatchSnapshot();
});

test("basic 15", async () => {
  expect(
    await run(`
      set $apple to "red"
      set $orange to "orange"
      log $apple + $orange
    `)
  ).toMatchSnapshot();
});

test("basic 16", async () => {
  expect(
    await run(`
      set global apple to "red"
      set global orange to "orange"
      log $apple + $orange
    `)
  ).toMatchSnapshot();
});

test("basic 17", async () => {
  expect(
    await run(`
    on "worldLoad"
        log "Hello!"
        set $h's b to event's a
    end
    `)
  ).toMatchSnapshot();
});
