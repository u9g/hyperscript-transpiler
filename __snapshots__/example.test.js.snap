// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`basic 1 1`] = `
"let it
const globals = {}
register("worldLoad", (my) => {
  exampleWorldLoad()

  exampleWorldLoad2()
})

register("quit", (my) => {
  sendLeave()
})
"
`;

exports[`basic 2 1`] = `
"let it
const globals = {}
register("click", (my) => {
  it = myJavascriptFunction()
})
"
`;

exports[`basic 3 1`] = `
"let it
const globals = {}
register("click", (my) => {
  it = prompt("age fn?")

  it = it()
})
"
`;

exports[`basic 4 1`] = `
"let it
const globals = {}
ChatLib.chat("Hello")

ChatLib.chat("World!")
"
`;

exports[`basic 5 1`] = `
"let it
const globals = {}
;(() => {
  ChatLib.chat("hello")(() => {
    ChatLib.chat("world")
    ChatLib.chat("!")
  })()
})()
"
`;

exports[`basic 6 1`] = `
"let it
const globals = {}
register("click", (my) => {
  ;(() => {
    ChatLib.chat("hello")
    ChatLib.chat("world")
  })()
})
"
`;

exports[`basic 7 1`] = `
"let it
const globals = {}
register("click", (my) => {
  ;(() => {
    it = helloWorld()
    ChatLib.chat(it)
  })()
})
"
`;

exports[`basic 8 1`] = `
"let it
const globals = {}
let x
x = "5"
"
`;

exports[`basic 9 1`] = `
"let it
const globals = {}
let x
x = apple
"
`;

exports[`basic 9 2`] = `
"let it
const globals = {}
let x
x = 45
"
`;

exports[`basic 10 1`] = `
"let it
const globals = {}
let x
x = { name: "Joe", age: 35 }
"
`;

exports[`basic 11 1`] = `
"let it
const globals = {}
register("keyup", (my) => {
  ChatLib.chat(it.x)
})
"
`;

exports[`basic 11 2`] = `
"let it
const globals = {}
;(() => {
  let y
  y = (x) => x
  it = y(5)
})()
"
`;

exports[`basic 12 1`] = `
"let it
const globals = {}
let y
y = () => true
"
`;

exports[`basic 13 1`] = `
"let it
const globals = {}
;(() => {
  let y
  y = { a: (x) => x }
  it = y.a("a")
})()
"
`;

exports[`basic 14 1`] = `
"let it
const globals = {}
let x, y, sumOfSquares
x = 10

y = 20

sumOfSquares = x * x + y * y
"
`;

exports[`basic 15 1`] = `
"let it
const globals = {}
globals.apple = "red"

globals.orange = "orange"

ChatLib.chat(globals.apple + globals.orange)
"
`;
